source/appModules/outlook.py

```diff
@@ -156,9 +164,15 @@ def event_NVDAObject_init(self,obj):
 			obj.role=controlTypes.ROLE_LISTITEM
 
 	def chooseNVDAObjectOverlayClasses(self, obj, clsList):
-		# Currently all our custom classes are IAccessible
 		if isinstance(obj,UIA) and obj.UIAElement.cachedClassName in ("LeafRow","ThreadItem","ThreadHeader"):
 			clsList.insert(0,UIAGridRow)

```

Comment:
I think you might one to return in this block, otherwise the logic under it is needlessly executed (i.e. fetching the role)